rules/crowdstrike_rules/crowdstrike_dns_request.py

```diff
@@ -8,10 +8,7 @@ def rule(event):
     # We need to run either for Crowdstrike.DnsRequest or for Crowdstrike.FDREvent with the
     # 'DnsRequest' fdr_event_type. Crowdstrike.DnsRequest is covered because of the
     # association with the type
-    if (
-        event.get("p_log_type") == "Crowdstrike.FDREvent"
-        and event.get("fdr_event_type", "") != "DnsRequest"
-    ):
+    if only_crowdstrike_fdr_event_type(event, 'DnsRequest'):
```

Comment:
Reconsidering my own suggestion here 😞 ... Since this doesn't read very well, we could potentially invert the boolean logic inside the helper function:

```python
def has_crowdstrike_fdr_event_type(event, name: str) -> bool:
    """
    Checks if the event has the Crowdstrike.FDREvent log type 
    and the event type equals the name parameter.
    """
    return not (
            event.get("p_log_type") == "Crowdstrike.FDREvent"
            and event.get("fdr_event_type", "") != name
    )
```

| `event` | `name` | Return Value |
| ---- | ---- | ---- |
| `{"p_log_type": "Crowdstrike.FDREvent", "fdr_event_type": "aid_master"}` | `DnsRequest` | False |
| `{"p_log_type": "Crowdstrike.FDREvent", "fdr_event_type": "DnsRequest"}` | `DnsRequest` | True |
| `{"p_log_type": "Crowdstrike.DNSRequest", "event_simpleName": "DnsRequest"}` | `DnsRequest` | True |

 
On the caller site:

```python
# Only process Crowdstrike.FDREvent events with type DnsRequest
if !has_crowdstrike_fdr_event_type(event, 'DnsRequest'):
    return False

Reply:
Good idea to write a truth table down. It'll be helpful. I actually wanted to avoid the helper function in general because, even with the best name, it can potentially cause confusion.
It doesn't hurt to have it though. Added: https://github.com/panther-labs/panther-analysis/pull/673/commits/8662a9c36a41f4ac82f7967e48738719edf32abf